From c8d40fb84c15c44e5433d8bef64b004bdfe676d3 Mon Sep 17 00:00:00 2001
From: Andrii Sultanov <andriy.sultanov@vates.tech>
Date: Thu, 8 Jan 2026 12:45:14 +0000
Subject: [PATCH] xenops: Fix migrate parameter ordering

localhost_migration was introduced as the last parameter for migrate in
xenops_interface.ml, but was second to last in xenops_server.ml.

This meant that non-localhost migrations with verify_dest=true were classed as
localhost and did not balloon down before migration.

Fixes: 091160444 ("xenopsd: Don't balloon down memory on same-host migration")

Signed-off-by: Andrii Sultanov <andriy.sultanov@vates.tech>
---
 ocaml/xenopsd/lib/xenops_server.ml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ocaml/xenopsd/lib/xenops_server.ml b/ocaml/xenopsd/lib/xenops_server.ml
index 6a06b36ba..16a3086d7 100644
--- a/ocaml/xenopsd/lib/xenops_server.ml
+++ b/ocaml/xenopsd/lib/xenops_server.ml
@@ -3826,7 +3826,7 @@ module VM = struct
   let s3resume _ dbg id = queue_operation dbg id (Atomic (VM_s3resume id))
 
   let migrate _context dbg id vmm_vdi_map vmm_vif_map vmm_vgpu_pci_map vmm_url
-      (compress : bool) (localhost_migration : bool) (verify_dest : bool) =
+      (compress : bool) (verify_dest : bool) (localhost_migration : bool) =
     let tmp_uuid_of uuid ~kind =
       Printf.sprintf "%s00000000000%c" (String.sub uuid 0 24)
         (match kind with `dest -> '1' | `src -> '0')
